identity-wallet/src/persistence.rs

```diff
@@ -199,3 +235,110 @@ pub fn persist_asset(file_name: &str, id: &str) -> Result<(), AppError> {
+pub fn apply_state_migrations(
+    app_state_object: serde_json::Map<String, serde_json::Value>,
+    mut current_version: u32,
+) -> anyhow::Result<AppState, AppError> {
+    while current_version < APP_STATE_VERSION {
+        // this code is commented out because we don't have any migrations yet.
+        // match current_version {
+        //     1 => migrate_v1_to_v2(&mut app_state_object)?,
+        //     _ => return Err(AppError::AppStateMigrationError(current_version, APP_STATE_VERSION, "Unsupported AppState version")),
+        // }
+
+        let previous_version = current_version;
+        // This is a temporary solution to avoid infinite loops.
+        current_version += 1;
+
+        // current_version = app_state_object
+        //     .get("version")
+        //     .and_then(|v| v.as_u64())
+        //     .map(|v| v as u32)
+        //     .ok_or_else(|| AppError::Error("Failed to get version while migrating AppState".to_string()))?;
+
+        debug!(
+            "state successfully migrated AppState version from {} to {}.",
+            previous_version, current_version
+        );
+    }
+
+    let app_state_value = serde_json::Value::Object(app_state_object);
+    let app_state: AppState = serde_json::from_value(app_state_value)?;
+
+    Ok(app_state)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        error::AppError,
+        state::{AppState, APP_STATE_VERSION},
+    };
+    use serde_json::{Map, Value};
+
+    // To unit test the appstate_version_update function, we need to create an adapted function.
+    // This is to bypass the need for a true STATE_FILE due to the fn `save_state`.
+    // Also to bypass migration functions, since we don't have multiple AppState versions yet.
+
+    pub async fn dummy_appstate_version_update(
+        mut app_state_object: Map<String, Value>,
+        mut outdated_version: u32,
+    ) -> anyhow::Result<AppState, AppError> {
+        while outdated_version < APP_STATE_VERSION {
+            match outdated_version {
+                0 => dummy_migrate_v0_to_v1(&mut app_state_object),
+                _ => {
+                    return Err(AppError::Error(format!(
+                        "Unsupported AppState version: {}",
+                        outdated_version
+                    )))
+                }
+            }
+
+            outdated_version = app_state_object
+                .get("version")
+                .and_then(|v| v.as_u64())
+                .map(|v| v as u32)
+                .ok_or_else(|| AppError::Error("Failed to get version while migrating AppState".to_string()))?;
+        }
+
+        let app_state_value = Value::Object(app_state_object);
+        let app_state: AppState = serde_json::from_value(app_state_value)?;
+
+        Ok(app_state)
+    }
+
+    fn dummy_migrate_v0_to_v1(app_state_object: &mut Map<String, Value>) {
+        app_state_object.insert("version".to_string(), Value::from(1));
+    }
```

Comment:
If you agree with [my comment above](https://github.com/impierce/identity-wallet/pull/498/files#r1975571962), then there is no need really for these dummy functions since you can just test `apply_state_migrations` directly.